hummingbot/connector/exchange/huobi/huobi_api_order_book_data_source.py

```diff
@@ -148,93 +86,109 @@ async def _subscribe_channels(self, ws: WSAssistant):
+        pass
+
+    def snapshot_message_from_exchange(self,
+                                       msg: Dict[str, Any],
+                                       metadata: Optional[Dict] = None) -> OrderBookMessage:
+
+        """
+        Creates a snapshot message with the order book snapshot message
+        :param msg: the response from the exchange when requesting the order book snapshot
+        :param timestamp: the snapshot timestamp
+        :param metadata: a dictionary with extra information to add to the snapshot data
+        :return: a snapshot message with the snapshot information received from the exchange
+        """
+        if metadata:
+            msg.update(metadata)
+        msg_ts = int(round(msg["tick"]["ts"] / 1e3))
+        msg_channel = msg["ch"]
+        content = {
+            "update_id": msg["tick"]["ts"],
+            "bids": msg["tick"]["bids"],
+            "asks": msg["tick"]["asks"]
+        }
+        if "trading_pair" in msg:
+            content["trading_pair"] = msg["trading_pair"]
+        else:
+            content["trading_pair"] = convert_from_exchange_trading_pair(msg_channel.split(".")[1])
+
+        return OrderBookMessage(OrderBookMessageType.SNAPSHOT, content, timestamp=msg_ts)
+
+    def trade_message_from_exchange(self,
+                                    msg: Dict[str, Any],
+                                    metadata: Dict[str, Any] = None) -> OrderBookMessage:
+        """
+        Creates a trade message with the information from the trade event sent by the exchange
+        :param msg: the trade event details sent by the exchange
+        :param metadata: a dictionary with extra information to add to trade message
+        :return: a trade message with the details of the trade as provided by the exchange
+        """
+        if metadata:
+            msg.update(metadata)
+
+        msg_ts = int(round(msg["ts"] / 1e3))
+        content = {
+            "trading_pair": msg["trading_pair"],
+            "trade_type": float(TradeType.BUY.value) if msg["direction"] == "buy" else float(TradeType.SELL.value),
+            "trade_id": msg["id"],
+            "update_id": msg["ts"],
+            "amount": msg["amount"],
+            "price": msg["price"]
+        }
+        return OrderBookMessage(OrderBookMessageType.TRADE, content, timestamp=msg_ts)
+
+    async def _process_websocket_messages(self, websocket_assistant: WSAssistant):
+        async for ws_response in websocket_assistant.iter_messages():
+            data: Dict[str, Any] = ws_response.data
+            if "ping" in data:
+                pong_request = WSJSONRequest(payload={"pong": data["ping"]})
+                await websocket_assistant.send(request=pong_request)
+                continue
+            channel: str = self._channel_originating_message(event_message=data)
+            if channel in [self._diff_messages_queue_key, self._trade_messages_queue_key]:
+                self._message_queue[channel].put_nowait(data)
```

Comment:
This should be replaced with a call to `tracked_order = self._order_tracker.all_updatable_orders.get(client_order_id)`.